### PR TITLE
DAOS-13146 test: Update the ENGINE_POOL_METRICS.

### PR DESCRIPTION
**Pull request by rpadma2**
Test-tag: telemetry_pool_metrics test_telemetry_pool_space_metrics tgt_dkey_akey_punch

Summary: Updating the Pool Metrics based on latest DAOS dmg telemetry metrics list command.

### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [x] You are the appropriate gatekeeper to be landing the patch.
* [x] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [x] Githooks were used. If not, request that user install them and check copyright dates.
* [x] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [x] All builds have passed.  Check non-required builds for any new compiler warnings.
* [x] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [x] If applicable, the PR has addressed any potential version compatibility issues.
* [x] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [x] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/telemetry/telemetry_pool_metrics.py [hunk 1]
```diff
@@ -187,3 +187,24 @@ def test_telemetry_pool_metrics(self):
...
+
+            Create a pool and check whether all the pool metrics listed
+            in the ENGINE_POOL_METRICS are valid.
+        Steps:
+            Create Pool
+            Get all the pool metrics and check for any errors.
+        :avocado: tags=all,daily_regression
+        :avocado: tags=vm
+        :avocado: tags=telemetry
+        :avocado: tags=telemetry_pool_metrics,test_telemetry_pool_metrics_sanity_check
```
Comment (daltonbohning) [inline src/tests/ftest/telemetry/telemetry_pool_metrics.py:202]: If you need to push again, please use the class name as a tag
```suggestion
        :avocado: tags=TelemetryPoolMetrics,test_telemetry_pool_metrics_sanity_check
```

File: src/tests/ftest/util/dmg_utils.py [hunk 2]
```diff
@@ -1340,9 +1340,9 @@ def get_json_response(data, key, description):
     """
     response = {}
     try:
-        if data['error']:
+        if 'error' in data and data['error']:
             raise CommandFailure("{} failed: {}".format(description, data['error']))
-        if len(data['response']['host_errors']) > 0:
+        if 'response' in data and len(data['response']['host_errors']) > 0:
```
Comment (daltonbohning) [inline src/tests/ftest/util/dmg_utils.py:1345]: Why is this necessary? If the json result doesn't contain `error` or `response`, isn't that a problem with the json returned by the `dmg` command? I.e. even if the command fails, the json should contain at least `error`

File: src/tests/ftest/telemetry/telemetry_pool_metrics.py (hunk 1 above)
Comment (rpadma2) [reply src/tests/ftest/telemetry/telemetry_pool_metrics.py:202]: If I need to push again, I will update the existing test and new test case.

File: src/tests/ftest/util/dmg_utils.py (hunk 2 above)
Comment (phender) [inline src/tests/ftest/util/dmg_utils.py:1345]: After discussing this with Dalton, we recommend:
```suggestion
        if 'host_errors' in data['response'] and len(data['response']['host_errors']) > 0:
```

**Review by phender — CHANGES_REQUESTED** [review]

**Review by shimizukko — APPROVED** [review]

File: src/tests/ftest/telemetry/telemetry_pool_metrics.py (hunk 1 above)
Comment (rpadma2) [reply src/tests/ftest/telemetry/telemetry_pool_metrics.py:202]: Done.

File: src/tests/ftest/util/dmg_utils.py (hunk 2 above)
Comment (rpadma2) [reply src/tests/ftest/util/dmg_utils.py:1345]: Done. the changes based on @phender and @daltonbohning .

Comment (rpadma2) [reply src/tests/ftest/util/dmg_utils.py:1345]: Done.

**Review by phender — APPROVED** [review]

**Comment by rpadma2** [comment]
All tests passed.

**Pull request merged**